cli/pub_command.go

```diff
@@ -129,6 +131,73 @@ func (c *pubCmd) prepareMsg(body []byte, seq int) (*nats.Msg, error) {
 	return msg, parseStringsToMsgHeader(c.hdrs, seq, msg)
 }
 
+func (c *pubCmd) doReqWait(nc *nats.Conn, progress *uiprogress.Bar) error {
```

Comment:
in order to decrease the duplication I think we can always use the same req - this one - we just need to know to exit after first message when wait duration is 0

The existing `doReqAsync()` function has a lot of this logic in it, including for example early short circuit when no responses are recieved for 300ms etc. 

So after @tbeets opened the issue my thought was that I would make that a bit more generic - or extract from it it's core 1:n reply handling to use here.  Thats a bit of a chore, happy to do that on Friday.

Reply:
I'll take a look, thanks @ripienaar.